src/Groups.jl

```diff
@@ -4,4 +4,256 @@
 #
 ###############################################################################
 
-elem_type(T::Type{<:Group}) = eltype(T)
+###############################################################################
+#
+#   Fundamental operations to be implemented for all group types
+#
+###############################################################################
+
+Base.one(G::Group) = throw(NotImplementedError(:parent, g))
+
+Base.parent(g::GroupElem) = throw(NotImplementedError(:parent, g))
+
+Base.:*(g::T, h::T) where {T<:GroupElem} = throw(NotImplementedError(:*, g, h))
+
+Base.inv(g::GroupElem) = throw(NotImplementedError(:inv, g))
+
+
+# also should implement:
+#   parent_type(::Type{MyGroupElem})
+#   elem_type(::Type{MyGroup})
+#
+# ... and many more of course (among those below and beyond), such as
+#   gens(::Group)
+#   ngens(::Group)
+#   order(::Group)
+#   order(::GroupElem)
+
+
+###############################################################################
+#
+#   Orders of groups and group elements
+#
+###############################################################################
+
+struct InfiniteOrderError{T} <: Exception
+    x::T
+    InfiniteOrderError(g::Union{GroupElem, Group}) = new{typeof(g)}(g)
+end
+
+function Base.showerror(io::IO, err::InfiniteOrderError{T}) where T
+    println(io, "Infinite order exception with ", err.x)
+    print(io, "order will only return a value when it is finite. ")
+    f = if T <: Group
+        "is_finite(G)"
+    elseif T <: GroupElem
+        "is_finiteorder(g)"
+    end
+    print(io, "You should check with `$f` first.")
+end
+
+@doc raw"""
+    order(::Type{T} = BigInt, G::Group) where T
+
+Return the order of $G$ as an instance of `T`. If $G$ is of infinite order,
+`InfiniteOrderError` exception will be thrown.
+
+!!! warning
+
+    Only arbitrary sized integers are required to return a mathematically
+    correct answer.
+"""
+function order(::Type{T}, G::Group) where T
+    throw(NotImplementedError(:order, G))
+end
```

Comment:
Throwing `function order(::Type{T}, G::GroupElem) where T` is missing. (For consistency with the one for Groups.)

Reply:
Added. I also revised the docstring for both `order` methods, please check.